### PR TITLE
Expression-mode for del, depri, do, pri, postpone.

### PR DESCRIPTION
**Pull request by mruwek**
Expression-mode is so great in `topydo edit` – why not have it also in other "multi commands"?

### CODE REVIEW

**Comment by mruwek** [comment]
I'll do the tests and push them here after initial review.

**Comment by bram85** [comment]
Haven't looked at it yet, but it's a good idea. However, I'd like to park this for a moment and release 0.4 first.

**Comment by mruwek** [comment]
> I'd like to park this for a moment and release 0.4 first.

Ok. I'll write some tests in the meantime.

File: topydo/commands/EditCommand.py [hunk 1]
```diff
@@ -73,12 +78,20 @@ def _todos_from_temp(self, p_temp_file):
 
         return todo_objs
 
-    def _open_in_editor(self, p_temp_file, p_editor):
+    def _open_in_editor(self, p_file):
```
Comment (bram85) [inline topydo/commands/EditCommand.py]: This change seems unrelated to the rest of the change. I'd suggest to put that in a different commit.

Comment (mruwek) [inline topydo/commands/EditCommand.py]: Done in 187e9ed,

File: test/DeleteCommandTest.py [hunk 2]
```diff
@@ -124,14 +128,14 @@ def test_multi_del1(self):
         command = DeleteCommand(["1", "2"], self.todolist, self.out, self.error, _no_prompt)
         command.execute()
 
-        self.assertEqual(self.todolist.count(), 0)
+        self.assertEqual(self.todolist.count(), 2)
```
Comment (bram85) [inline test/DeleteCommandTest.py]: Checking just the count is not enough, also check the content of what's left.

File: test/DeleteCommandTest.py [hunk 3]
```diff
@@ -124,14 +128,14 @@ def test_multi_del1(self):
...
 
     def test_multi_del2(self):
         """ Test deletion of multiple items. """
         command = DeleteCommand(["1", "2"], self.todolist, self.out, self.error, _yes_prompt)
         command.execute()
 
-        self.assertEqual(self.todolist.count(), 0)
+        self.assertEqual(self.todolist.count(), 2)
```
Comment (bram85) [inline test/DeleteCommandTest.py]: Same: checking just the count is not enough, also check the content of what's left.

File: test/DeleteCommandTest.py [hunk 4]
```diff
@@ -160,6 +164,47 @@ def test_multi_del5(self):
...
+
+        self.assertFalse(self.todolist.is_dirty())
+
+    def test_expr_del4(self):
+        """ Remove only relevant todo items. """
+        command = DeleteCommand(["-e", ""], self.todolist, self.out, self.error, None)
+        command.execute()
+
+        self.assertTrue(self.todolist.is_dirty())
+        self.assertEqual(self.todolist.count(), 1)
```
Comment (bram85) [inline test/DeleteCommandTest.py]: Same: checking just the count is not enough, also check the content of what's left.

**Comment by bram85** [comment]
I see there's a test case missing for `EditCommand`: the case where no arguments are passed. All other classes seem to be covered by the tests.

I will merge in the `coverage` branch after this PR. Later on I'll probably also switch from statement coverage to branch coverage.

The reason I already bothered to run `coverage` on this is that I
